docs/maintainer.md

```diff
@@ -0,0 +1,212 @@
+
+A series base can be found with the command
+
+    git merge-base remote/series_branch_or_tag origin/master
+
+
+### Applying patches and patch series from the mailing list
+
+Before you begin, check out the `next` branch and pull
+from scylla.git to be up to date. If the pull isn't clean,
+abort the merge and use `git pull --rebase`. Examine the result
+to see if you forgot to push a previously applied patch.
+
+### Applying single patches from the mailing list
+
+Save the patch(es) to some directory, and use the command
+
+    git am -im3 /path/to/patches/*.eml
+
+to apply the patches. `-i` makes the process interactive and
+lets you edit the commit message, `-m` sets the Message-Id
+tag (which is used by Commit Bot to set the Reply-To header,
+so that the commit acknowledgement appears as a response to
+the patch), and `-3` enables 3-way merging which is rumored
+to reduce conflicts.
+
+Use `git push` to publish the patches.
+
+### Applying patch series from the mailing list
+
+Indentify the git url and branch/tag identifier, and issue
+the command
+
+    git pull --log --no-ff <url> <branch/tag>
+
+The `--log` flag generates a list of patches in the commit log,
+while `--no-ff` ensures a merge commit is created.
+
+Copy the cover letter subject and body to the merge commit's
+subject and body, respectively. Make sure the merge commit
+supplies enough information to understand what the series is
+doing without having to read individual commits.
+
+Use `git push` to publish the patches.
+
+### Applying patches and patch series from github pull requests
+
+A common contributor mistake is to base patches on `next`
+rather than on `master`. This results in random commits
+appearing in the pull request, if `next` is edited for some
+reason.
+
+Ensure that the target branch in the pull request page is
+set to `scylladb:next` (click Edit and change it if that's
+not the case).
+
+When merging, verify that github didn't mangle the commit
+log, check both your and the contributor's email address
+are correct (no @noreply.users.github.com or similar
+address, or home vs. work addresses).
+
+### Applying single patches from github pull requests
+
+Select "Squash and merge" and follow through.
+
+### Applying patch series from github pull requests
+
+Select "Create a merge commit" and follow through.
+
+## Dequeuing bad patches
+
+Sometimes, a patch fails promotion by Jenkins, or needs to
+be dequeued for some other reason. This section explains how.
+
+ 1. Synchronize with origin by checking out `next` and
+    issuing a `git pull`
+ 2. Issue `git rebase -i --rebase-merges origin/master`
+ 3. Identify the final section that contains the pick/merge
+    command that will contain the result. Ignore any intermediate
+    sections that describe branches.
+ 4. Delete pick/merge commands that correspond to bad commits
+ 5. Save the file and let `git rebase` do the work
+ 6. Publish your changes with `git push --force-with-leases`
```

Comment:
there's a minor typo here - it's `force-with-lease` - it's not that important though, since it's impossible to confuse `force-with-lease` with any other parameter